### PR TITLE
v1.15 Backports 2024-05-30

### PR DESCRIPTION
**Pull request by joamaki**
* [x] #32428 (@julianwiedmann) :warning: resolved conflicts
 * [x] #32654 (@giorio94) :warning: resolved conflicts
 * [x] #32684 (@giorio94) :warning: resolved conflicts
 * [x] #32725 (@gandro) :warning: resolved conflicts
 * [x] #32746 (@YutaroHayakawa)
 * [x] #31671 (@foyerunix) :warning: resolved conflicts

Fixed minor conflicts due to some diff lines not being present in v1.15.

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 32428 32654 32684 32725 32746 31671
```

### CODE REVIEW

**Comment by joamaki** [comment]
/test-backport-1.15

**Comment by julianwiedmann** [comment]
Could you drop https://github.com/cilium/cilium/pull/32321 please? It was meant as a pre-req to make another backport easier, but that ended up requiring a custom PR anyway (https://github.com/cilium/cilium/pull/32778).

Sorry for the trouble :/.

**Comment by joamaki** [comment]
> Could you drop #32321 please? It was meant as a pre-req to make another backport easier, but that ended up requiring a custom PR anyway (#32778).
> 
> Sorry for the trouble :/.

Done.

**Comment by joamaki** [comment]
/test-backport-1.15

File: bpf/tests/lib/egressgw.h [hunk 1]
```diff
@@ -209,47 +209,50 @@ static __always_inline int egressgw_snat_check(const struct __ctx_buff *ctx,
...
-			   test_ctx.tx_packets, ct_entry->tx_packets)
-	if (ct_entry->rx_packets != test_ctx.rx_packets)
-		test_fatal("bad RX packet count (expected %u, actual %u)",
-			   test_ctx.rx_packets, ct_entry->rx_packets)
+	  if (ct_entry->tx_packets != test_ctx.tx_packets)
+	  	test_fatal("bad TX packet count (expected %u, actual %u)",
+	  		   test_ctx.tx_packets, ct_entry->tx_packets)
+	  if (ct_entry->rx_packets != test_ctx.rx_packets)
+	  	test_fatal("bad RX packet count (expected %u, actual %u)",
+	  		   test_ctx.rx_packets, ct_entry->rx_packets)
```
Comment (julianwiedmann) [inline bpf/tests/lib/egressgw.h]: This is where you hit the conflict, right? The resolution looks good (keep the old packet count checks), but the indentation looks slightly off.

**Review by giorio94 — APPROVED** [review]
My commits look good, thanks!

**Review by sayboras — APPROVED** [review]
Looks good for #31671

File: bpf/tests/lib/egressgw.h (hunk 1 above)
Comment (joamaki) [reply bpf/tests/lib/egressgw.h]: What do you mean slightly off? It's completely off! Fixed ;)

**Review by brlbil — APPROVED** [review]

**Review by julianwiedmann — APPROVED** [review]
thank you!

**Comment by giorio94** [comment]
/test-backport-1.15

**Review by YutaroHayakawa — APPROVED** [review]

**Review by joestringer — APPROVED** [review]

**Pull request merged**